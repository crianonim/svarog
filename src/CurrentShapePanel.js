import React from 'react';
import BasicAttrEditor from './BasicAttrEditor';

const CurrentShapePanel = ({shape,changed})=>{
    console.log({shape},{changed})
    return (
        <>
        <BasicAttrEditor
          element={shape.shape}
          attrs={shape.attributes}
          changed={attrs => {
            shape.attributes=attrs;
            changed(shape);
          }}
        />
        </>
    )
}

export default CurrentShapePanel;
