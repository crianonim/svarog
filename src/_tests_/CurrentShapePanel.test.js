import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CurrentShapePanel from "../components/CurrentShapePanel";

it("matches a snapshrot",()=>{
    const wrapper=shallow(<CurrentShapePanel shape={{element:"circle",attributes:{"x":10,"y":20,"r":30}} } />) 
    expect(toJson(wrapper)).toMatchSnapshot();
});
