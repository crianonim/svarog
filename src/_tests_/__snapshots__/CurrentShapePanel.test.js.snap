// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshrot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-heading"
  >
    Edit properties of the selected 
    <b />
  </div>
  <div
    className="panel-block"
  >
    <BasicAttrEditor
      attrs={
        Object {
          "r": 30,
          "x": 10,
          "y": 20,
        }
      }
      changed={[Function]}
    />
  </div>
</div>
`;
